File: src/CANON_SDK/EDSDK_v13.19.0_Macintosh/Sample/Objective-C/CameraControl/command/OpenSessionCommand.m
/******************************************************************************
*                                                                             *
*   PROJECT : EOS Digital Software Development Kit EDSDK                      *
*      NAME : OpenSessionCommand.m                                            *
*                                                                             *
*   Description: This is the Sample code to show the usage of EDSDK.          *
*                                                                             *
*                                                                             *
*******************************************************************************
*                                                                             *
*   Written and developed by Camera Design Dept.53                            *
*   Copyright Canon Inc. 2007-2010 All Rights Reserved                        *
*                                                                             *
*******************************************************************************
*   File Update Information:                                                  *
*     DATE      Identify    Comment                                           *
*   -----------------------------------------------------------------------   *
*   06-03-22    F-001        create first version.                            *
*                                                                             *
******************************************************************************/

#import "OpenSessionCommand.h"
#import "EDSDK.h"
#import "cameraEvent.h"
#include "GetPropertyCommand.h"
#include "GetPropertyDescCommand.h"

@implementation OpenSessionCommand

-(BOOL)execute
{
	EdsError error = EDS_ERR_OK;
	CameraEvent * event;
	NSNumber * number;
	EdsUInt32 saveTo;
//	GetPropertyCommand * getPropertyCommand;
//	GetPropertyDescCommand * getPropertyDescCommand;

	//The communication with the camera begins
	error = EdsOpenSession([_model camera]);
	
    //Preservation ahead is set to PC
    if(error == EDS_ERR_OK)
    {
        saveTo = kEdsSaveTo_Host;
        error = EdsSetPropertyData([_model camera], kEdsPropID_SaveTo, 0, sizeof(saveTo) , &saveTo);
    }
		
			
    if(error == EDS_ERR_OK)
    {
        EdsCapacity capacity = {0x7FFFFFFF, 0x1000, 1};
        error = EdsSetCapacity([_model camera], capacity);
    }

	//Notification of error
	if(error != EDS_ERR_OK)
	{
		number = [[NSNumber alloc] initWithInt: error];
		event = [[CameraEvent alloc] init:@"error" withArg: number];
		[_model notifyObservers:event];	
		[event release];
		[number release];
	}
	
	return YES;
}

@end

File: src/CANON_SDK/EDSDK_v13.19.0_Windows/sample/VB/CameraControl/Command/OpenSessionCommand.vb
'/******************************************************************************
'*                                                                             *
'*   PROJECT : EOS Digital Software Development Kit EDSDK                      *
'*      NAME : OpenSessionCommand.vb                                           *
'*                                                                             *
'*   Description: This is the Sample code to show the usage of EDSDK.          *
'*                                                                             *
'*                                                                             *
'*******************************************************************************
'*                                                                             *
'*   Written and developed by Camera Design Dept.53                            *
'*   Copyright Canon Inc. 2006 All Rights Reserved                             *
'*                                                                             *
'*******************************************************************************
'*   File Update Information:                                                  *
'*     DATE      Identify    Comment                                           *
'*   -----------------------------------------------------------------------   *
'*   06-03-22    F-001        create first version.                            *
'*                                                                             *
'******************************************************************************/

Option Explicit On
Imports System.Runtime.InteropServices

Public Class OpenSessionCommand
    Inherits Command

    Public Sub New(ByVal model As CameraModel)
        MyBase.new(model)
    End Sub

    '// Execute a command.	
    Public Overrides Function execute() As Boolean

        Dim err As Integer = EDS_ERR_OK
        Dim locked As Boolean = False

        '// Open session with remote camera.
        err = EdsOpenSession(MyBase.model.getCameraObject())


        'Preservation ahead is set to PC
        If err = EDS_ERR_OK Then

            Dim saveTo As Integer = EdsSaveTo.kEdsSaveTo_Host
            err = EdsSetPropertyData(MyBase.model.getCameraObject(), kEdsPropID_SaveTo, 0, Marshal.SizeOf(saveTo), saveTo)

        End If



        If err = EDS_ERR_OK Then

            Dim capacity As EdsCapacity
            capacity.numberOfFreeClusters = &H7FFFFFFF
            capacity.bytesPerSector = &H1000
            capacity.reset = 1

            err = EdsSetCapacity(MyBase.model.getCameraObject(), capacity)

        End If




        'Notification of error
        If err < EDS_ERR_OK Then

            'CameraEvent e("error", &err);
            MyBase.model.notifyObservers(errr, err)

        End If


        Return True
    End Function

End Class
